### PR TITLE
Add  and update to recommandeded practices

### PR DESCRIPTION
**Pull request by liamqma**
- Add `extractstylestodirectory`
- Update some code snippets to our recommended practices.

### CODE REVIEW

**Comment by liamqma** [comment]
Merge this PR after https://github.com/atlassian-labs/compiled/pull/1513 is merged.

File: packages/docs/src/pages/writing-css.mdx [hunk 1]
```diff
@@ -6,17 +6,17 @@ order: 4
...
 
-Write CSS using template literals and objects with any of the APIs available with Compiled.
+Write CSS using object styles and objects with any of the APIs available with Compiled.
 
 ```jsx
 import { styled } from '@compiled/react';
 
-const Button = styled.button`
+const Button = styled.button({
   font-size: 10px;
```
Comment (dddlr) [inline packages/docs/src/pages/writing-css.mdx:15]: `font-size: 10px;` should be `fontSize: '10px',`

and so on for the other examples

File: packages/docs/src/pages/writing-css.mdx [hunk 2]
```diff
@@ -66,14 +68,14 @@ In the example below the two selectors (textarea and input) both alter the color
 
 ```jsx
 <div
-  css={{
+  css={css({
     'textarea,input': {
```
Comment (dddlr) [inline packages/docs/src/pages/writing-css.mdx:72]: nit: `textarea,input` could be changed to `textarea, input` (with a space after the comma) for better readability maybe 🤔

**Review by dddlr — CHANGES_REQUESTED** [review]

File: packages/docs/src/pages/writing-css.mdx (hunk 1 above)
Comment (liamqma) [reply packages/docs/src/pages/writing-css.mdx:15]: good catch!

**Review by dddlr — APPROVED** [review]
LGTM, thanks Liam! ❤️

**Pull request merged**